### PR TITLE
Improve bean-matcher for lists and arrays.

### PR DESCRIPTION
**Pull request by reitzmichnicht**
Adds a ListGenerator and compares arrays by elements,
not by reference.
Increased required JDK to 1.7 for null safe equals.

### CODE REVIEW

**Comment by reitzmichnicht** [comment]
This change would be a great improvement because a lot of static code checks require getters to return copies of lists or arrays so the internal state of a bean can not be modified from the outside.
I did not find a spec for beans that would require the getter to return the same reference as the setter.

**Comment by reitzmichnicht** [comment]
Any chance to get this merged?

**Comment by orien** [comment]
Thanks for your contribution!

File: src/main/java/com/google/code/beanmatchers/ListGenerator.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.google.code.beanmatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class ListGenerator implements ValueGenerator<List> {
+
+  public List generate() {
+    return new ArrayList();
```
Comment (ojchase) [inline src/main/java/com/google/code/beanmatchers/ListGenerator.java:9]: `ValueGenerator` requires repeated calls to the generator to return items that have are not equal per the `equals` method. Violating this introduces #21 .

**Pull request merged**